File: arrays/prob1122.py
# 1122 Relative sort array

class Solution(object):
    def relativeSortArray(self, arr1, arr2):
        """
        :type arr1: List[int]
        :type arr2: List[int]
        :rtype: List[int]
        """
        wc = {}
        for n in arr1:
            wc[n] = wc.get(n, 0) + 1
        ans = []
        for n in arr2:  # relative order in arr2 kept
            ans.extend([n]*wc[n])
            del wc[n]
        for n in sorted(wc.keys()): # tail sorted too
            ans.extend([n]*wc[n])
        return ans

File: hashtable/prob1122.kt
/* 1122 Relative Sort Array*/
class Solution {

    fun relativeSortArray(arr1: IntArray, arr2: IntArray): IntArray {

        val m = mutableMapOf<Int,Int>()
        for (v in arr1) {
            m[v] = m.getOrDefault(v, 0) + 1
        }

        var ans = mutableListOf<Int>()

        for (v in arr2) {
            ans.addAll(List<Int>(m[v]!!) {it -> v})
            m.remove(v)
        }

        val tmp = m.toList().sortedBy {it -> it.first}
        for ((k, v) in tmp) {
            ans.addAll( List(v) {k})
        }

        return ans.toIntArray()

    }
}
